test/test_dynamic_shape_models.py

```diff
@@ -117,11 +124,29 @@ def test_backward_pass_with_dynamic_input(self):
       xm.mark_step()
     print('Test passed.')
 
-  def create_dynamic_test_data(self, num_test_samples, num_features, device):
-    x_test = torch.ones(num_test_samples, num_features)
-    x_test[0][0] = 0
-    y_test = torch.ones(num_test_samples * 2)
-    y_test[0] = 0
+  def create_dynamic_test_data(self,
+                               num_test_samples,
+                               num_features,
+                               device,
+                               num_non_zeros=1):
+    x_test = torch.zeros(num_test_samples, num_features)
+    num_non_zero_added = 0
+    for i in range(num_test_samples):
+      for j in range(num_features):
+        x_test[i][j] = 1
+        num_non_zero_added += 1
+        if num_non_zero_added == num_non_zero_added:
```

Comment:
What does this condition do? Is it supposed to be always True?

Reply:
oops, fixed.